### PR TITLE
fix: [2U-607] reworked Grading Deadline section

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
### **Description**
- reworked Grading Deadline section
- removed translations copies

[YouTrack](https://youtrack.raccoongang.com/agiles/104-1121/105-3071?issue=2U-607)

### CODE REVIEW

File: src/grading-settings/deadline-section/messages.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ const messages = defineMessages({
     id: 'course-authoring.grading-settings.deadline.description',
     defaultMessage: 'Leeway on due dates',
   },
+  gracePeriodOnDeadlineErrorMsg: {
+    id: 'course-authoring.grading-settings.deadline.error.message',
+    defaultMessage: 'Grace period must be specified in HH:MM format.',
```
Comment (ruzniaievdm) [inline src/grading-settings/deadline-section/messages.js:14]: please get it from `constants.js` above

File: src/grading-settings/deadline-section/utils.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * Validates inputStr as a time in HH:MM format.
+ *
+ * @param {string} inputStr - the input string to validate.
+ * @param {function} setShowSavePrompt - a function to control save prompt display.
+ * @param {function} setIsError - a function to set error state.
+ * @returns {boolean} - returns `true` if `inputStr` is a valid time, else `false`.
+ */
+export function timerValidation(inputStr, setShowSavePrompt, setIsError) {
+  const timePattern = /^\d{2,3}:\d{2}$/;
+  const EMPTY_STR = '';
```
Comment (ruzniaievdm) [inline src/grading-settings/deadline-section/utils.js:21]: take into account comment above

File: src/grading-settings/deadline-section/utils.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Validates inputStr as a time in HH:MM format.
+ *
+ * @param {string} inputStr - the input string to validate.
+ * @param {function} setShowSavePrompt - a function to control save prompt display.
+ * @param {function} setIsError - a function to set error state.
+ * @returns {boolean} - returns `true` if `inputStr` is a valid time, else `false`.
+ */
+export function timerValidation(inputStr, setShowSavePrompt, setIsError) {
```
Comment (ruzniaievdm) [inline src/grading-settings/deadline-section/utils.js:19]: please try something like this
```
export function timerValidation(inputStr, setShowSavePrompt, setIsError) {
  const timePattern = /^\d{2,3}:\d{2}$/;

  const isValid = timePattern.test(inputStr);
  setShowSavePrompt(isValid);
  setIsError(!isValid);

  return isValid;
}
```

File: src/i18n/messages/uk.json [hunk 4]
```diff
@@ -790,13 +788,6 @@
   "course-authoring.course-rerun.sidebar.section-3.title": "What does not transfer from the original course?",
   "course-authoring.course-rerun.sidebar.section-3.description": "You are the only member of the new course's staff. No students are enrolled in the course, and there is no student data. There is no content in the discussion topics or wiki.",
   "course-authoring.course-rerun.sidebar.section-4.link": "Learn more about course re-runs",
-  "course-authoring.course-outline.sidebar.section-4.link": "Learn more about content visibility settings",
```
Comment (ruzniaievdm) [inline src/i18n/messages/uk.json:793]: why did you remove this diff only in this file?

**Review by ruzniaievdm — CHANGES_REQUESTED** [review]

File: src/grading-settings/deadline-section/index.jsx [hunk 5]
```diff
@@ -1,46 +1,66 @@
...
-      ...prevData,
-      gracePeriod: {
-        hours: Number(hoursAndMinutes[0]),
-        minutes: parseInt(hoursAndMinutes[1] ?? 0, 10),
-      },
-    }));
-    setShowSuccessAlert(false);
+    const { value } = e.target;
+    const [hours, minutes] = value.split(':');
+    const EMPTY_STR = ' ';
```
[the remark this answers is not in the document]
Comment (PKulkoRaccoonGang) [reply src/grading-settings/deadline-section/index.jsx:26]: These constants were different, now one of them has been removed. Let's leave this constant here, I came up with a more appropriate name for it

**Pull request closed without merging**